src/coordination/coordinator_cluster_state.cpp

```diff
@@ -99,64 +108,37 @@ auto CoordinatorClusterState::DoAction(TRaftLog log_entry, RaftLogAction log_act
   }
 }
 
-// TODO: (andi) Improve based on Gareth's comments
 auto CoordinatorClusterState::Serialize(ptr<buffer> &data) -> void {
   auto lock = std::shared_lock{log_lock_};
-  auto const role_to_string = [](auto const &role) -> std::string_view {
-    switch (role) {
-      case ReplicationRole::MAIN:
-        return "main";
-      case ReplicationRole::REPLICA:
-        return "replica";
-    }
-  };
 
-  auto const entry_to_string = [&role_to_string](auto const &entry) {
-    return fmt::format("{}_{}", entry.first, role_to_string(entry.second.role));
-  };
+  // .at(0) is hack to solve the problem with json serialization of map
+  auto const log = nlohmann::json{instances_}.at(0).dump();
```

Comment:
I would change this to be without braces, we could have a bug if we move to GCC or a different Clangd version
https://json.nlohmann.me/home/faq/#brace-initialization-yields-arrays

Reply:
Changing it here https://github.com/memgraph/memgraph/pull/1788/files#